x-pack/plugins/security_solution/public/timelines/components/timeline/body/column_headers/index.test.tsx

```diff
@@ -307,4 +253,26 @@ describe('ColumnHeaders', () => {
       expect(wrapper.exists('[data-test-subj="test-header-action-cell"]')).toBeTruthy();
     });
   });
+
+  describe('Fields', () => {
+    test('Closes field editor when the timeline is closed', () => {
+      const mockCloseEditor = jest.fn();
+      mockUseCreateFieldButton.mockImplementation((_, __, fieldEditorActionsRef) => {
+        fieldEditorActionsRef.current = { closeEditor: mockCloseEditor };
+        return <></>;
+      });
+
+      const wrapper = mount(
+        <TestProviders>
+          <ColumnHeadersComponent {...defaultProps} />
+        </TestProviders>
+      );
+      wrapper.setProps({ ...defaultProps, show: false });
+
+      // let useEffect hook execute
+      setTimeout(() => {
+        expect(mockCloseEditor).toHaveBeenCalled();
+      }, 0);
```

Comment:
@semd I don't understand how it is working. You are calling `setProps` on `TestProviders ` and not on `ColumnHeadersComponent `. Because you can only call `setProps` on the root people usually create a [Proxy component](https://github.com/enzymejs/enzyme/issues/1925#issuecomment-490637648) for this scenario. Here is an example on security solutions repo: https://github.com/elastic/kibana/blob/af8d5eb4e6b8e13d5c57834c0cfcce1c6491c596/x-pack/plugins/security_solution/public/common/components/query_bar/index.test.tsx#L24 

```suggestion
      const Proxy = (props: ColumnHeadersComponentProps) => (
        <TestProviders>
          <ColumnHeadersComponent {...props} />
        </TestProviders>
      );
      const wrapper = mount(<Proxy {...defaultProps} />);

      wrapper.setProps({ show: false });

      expect(mockCloseEditor).toHaveBeenCalled();
```

Reply:
true! very weird this was passing 😲